ape_safe/accounts.py

```diff
@@ -36,32 +47,94 @@ def aliases(self) -> Iterator[str]:
         for p in self._account_files:
             yield p.stem
 
-    def __len__(self) -> int:
-        return len([*self._account_files])
-
     @property
     def accounts(self) -> Iterator[AccountAPI]:
         for account_file in self._account_files:
-            yield SafeAccount(account_file_path=account_file)  # type: ignore
+            yield SafeAccount(account_file_path=account_file)
+
+    def __len__(self) -> int:
+        return len([*self._account_files])
+
+    def __setitem__(self, alias: str, address: str):
+        self.save_account(alias, address)
+
+    def __delitem__(self, alias: str):
+        self.delete_account(alias)
 
     def save_account(self, alias: str, address: str):
         """
         Save a new Safe to your ape configuration.
+
+        Args:
+            alias (str): The alias to save the Safe under.
+            address (str): The address of the Safe account.
         """
         chain_id = self.provider.chain_id
         account_data = {"address": address, "deployed_chain_ids": [chain_id]}
-        path = self.data_folder.joinpath(f"{alias}.json")
+        path = self._get_path(alias)
+        if path.is_file():
+            raise ApeSafeError(f"Safe with alias '{alias}' already exists.")
+
         path.write_text(json.dumps(account_data))
 
     def load_account(self, alias: str) -> "SafeAccount":
-        account_path = self.data_folder.joinpath(f"{alias}.json")
+        """
+        Load the Safe account.
+
+        Args:
+            alias (str): The alias the Safe account is saved under.
+
+        Returns:
+            ``:class:~ape_safe.accounts.SafeAccount``: The Safe account loaded.
+        """
+        account_path = self._get_path(alias)
+        if not account_path.is_file():
+            raise ApeSafeError(f"Safe with '{alias}' does not exist")
+
         return SafeAccount(account_file_path=account_path)
 
     def delete_account(self, alias: str):
-        path = self.data_folder.joinpath(f"{alias}.json")
+        """
+        Delete the local Safe account.
+        **NOTE**: If the account does not exist, nothing happens.
+
+        Args:
+            alias (str): The alias the Safe account is saved under.
+        """
+        self._get_path(alias).unlink(missing_ok=True)
+
+    def _get_client(self, key: str) -> BaseSafeClient:
+        if key in self.aliases:
+            safe = self.load_account(key)
+            return safe.client
+
+        elif key in self:
+            account = self[key]  # type: ignore[index]
```

Comment:
**question**: Is this type ignore because key is not `AddressType`? If yes, using `cast` is a better solution:

```python
account = self[cast(AddressType, key)]
```

else, ignore this comment!